org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -1336,6 +1337,15 @@ public Preferences setSettingsUrl(String settingsUrl) {
 
 	public Preferences setResourceFilters(List<String> resourceFilters) {
 		this.resourceFilters = resourceFilters == null ? new ArrayList<>() : resourceFilters;
+		if (resourceFilters != null) {
+			String argument = String.join("|", resourceFilters);
+			try {
+				Pattern.compile(argument);
```

Comment:
What about check the elements in the list one by one and only exclude invalid ones?

Reply:
Fixed.